src/http.ts

```diff
@@ -32,6 +37,7 @@ export interface ConnectionOptions {
 }
 
 // The trick to typecast union type alias
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
 function isSubsetErrorCode<T>(code: any, messages: { [c: number]: string }): code is T {
```

Comment:
Yeah I know, and I tried to fix, but because it is a guard trying to check if the code is a valid key in the messages object TS will complain if it's not `any`, i.e.:
> A type predicate's type must be assignable to its parameter's type.
  Type 'T' is not assignable to type 'number'.